rag_app/models.py

```diff
@@ -64,3 +64,12 @@ class EvaluationDataItem(BaseModel):
     answer: str
     chunk: str
     chunk_id: str
+
+
+class KeywordExtractionResponse(BaseModel):
+    keywords: List[str] = Field(
+        ...,
+        title="Keywords",
+        description="A list of at least 3 relevant keywords extracted from the question.",
```

Comment:
can you try to improve this prompt

"say, the goal is to render a range of keywords, expand acryonms used synonoms etc etc

Reply:
I played around with the prompt a bit for fts, got a small improvement

```
              Mean Values              
┏━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┓
┃ Metric         ┃ Value              ┃
┡━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━┩
│ MRR@3          │ 0.6198             │
│ MRR@5          │ 0.6328             │
│ MRR@10         │ 0.642              │
│ MRR@20         │ 0.6478             │
│ NDCG@3         │ 0.653469387755102  │
│ NDCG@5         │ 0.6781632653061224 │
│ NDCG@10        │ 0.6993877551020409 │
│ NDCG@20        │ 0.7204081632653062 │
│ retrieved_size │ 17.12              │
```

This is the semantic

```
┏━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━┓
┃ Metric         ┃ Value              ┃
┡━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━┩
│ MRR@3          │ 0.8031999999999999 │
│ MRR@5          │ 0.8131999999999999 │
│ MRR@10         │ 0.8131999999999999 │
│ MRR@20         │ 0.8162             │
│ NDCG@3         │ 0.8281999999999999 │
│ NDCG@5         │ 0.8453999999999999 │
│ NDCG@10        │ 0.8453999999999999 │
│ NDCG@20        │ 0.8558             │
│ retrieved_size │ 25.0               │
```